### ISSUE TITLE
format-imports@2.4.5 ignoring eslint semi-colon rule

### ISSUE THREAD

**Issue by aaronmccall** [daidodo/format-imports#12]
**Describe the bug**
When I run format-imports, it is adding semi-colons to import statements despite eslint rule disallowing them.

**To Reproduce**
Is there a demo repo to reproduce the issue?
Yes
Link: https://github.com/aaronmccall/format-imports-semi

1. clone repo
2. run npm i && npm run repro
3. note that after format-imports runs semi-colons are added and after eslint --fix runs they are removed

**Expected behavior**
When .eslintrc contains `"semi": ["error", "never"]`, format-imports does not insert semi-colons.

**Actual behavior**
Despite .eslintrc containing `"semi": ["error", "never"]`, format-imports inserts semi-colons.

**OS (please complete the following information):**

- OS: MacOS
- Version: 10.15.7 (Catalina)

**Editor (please complete the following information):**

- Editor: VSCode
- Version: 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3)

**package.json (please remove any sensitive info)**

```json
{
  "name": "format-imports-semi",
  "version": "1.0.0",
  "description": "reproduces a semi-colon bug between format-imports@2.4.5 and eslint@8.6.0",
  "repository": "git@github.com:aaronmccall/format-imports-semi.git",
  "license": "The Unlicense",
  "private": true,
  "type": "module",
  "scripts": {
    "sort-imports": "yarn format-imports -e jsx src",
    "lint": "yarn eslint --cache --ext=jsx --fix src",
    "repro": "yarn sort-imports && echo \"Diff after format-imports\" && git --no-pager diff && yarn lint && echo \"Diff after eslint --fix\" && git --no-pager diff"
  },
  "devDependencies": {
    "eslint": "~8.6.0",
    "eslint-plugin-import": "^2.24.2",
    "eslint-plugin-react": "~7.28.0",
    "format-imports": "^2.4.5"
  },
  "dependencies": {
    "react": "^17.0.0",
    "react-dom": "^17.0.0"
  },
  "importerSorter": {
    "groupRules": [
      {
        "regex": "^react$"
      },
      {},
      {
        "regex": "^[.]"
      }
    ],
    "hasSemicolon": false,
    "wrappingStyle": {
      "maxBindingNamesPerLine": 3,
      "maxDefaultAndBindingNamesPerLine": 4
    },
    "maxLineLength": 100,
    "trailingComma": "multiLine"
  }
}
```

**.eslintrc.\* (if any)**

```json
{
  "env": {
    "browser": true,
    "es2020": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ],
  "parserOptions": {
    "ecmaVersion": 11,
    "sourceType": "module"
  },
  "plugins": ["react"],
  "rules": {
    "camelcase": ["error"],
    "no-unused-vars": ["error"],
    "react/prop-types": ["warn", {
      "skipUndeclared": true
    }],
    "semi": ["error", "never"]
  },
  "settings": {
    "react": {
      "pragma": "React",
      "fragment": "Fragment",
      "version": "detect"
    }
  }
}

**Comment by daidodo** [comment]
@aaronmccall 
Thanks for the feedback!

The plugin doesn't support this rule yet.
But I'll see what I can do here.

**Comment by daidodo** [comment]
And I also found a few typo/improvements in your *package.json*:
* The config should be `importSorter` instead of `importerSorter` for this plugin.
* If you fix the above, the semicolon will be gone as you set `"hasSemicolon": false`. And other settings will be in effect.
* The `groupRules` can be simplified to `"groupRules": ["^react$", {}, "^[.]"]`. See [wiki](https://github.com/daidodo/format-imports/wiki/Grouping-Rules#abbreviation).

I'm going to support ESLint [semi](https://eslint.org/docs/rules/semi) rule in the next version. By then you won't need `"hasSemicolon": false` any more.

**Comment by daidodo** [comment]
I've released [v7.4.6](https://marketplace.visualstudio.com/items?itemName=dozerg.tsimportsorter) to support ESLint semi rule.
With it, you can remove `"hasSemicolon": false` because the plugin will set it automatically.

Please take a shot and tell me if you have more questions.
Thanks!

**Comment by aaronmccall** [comment]
Works great.

Thanks for the help!
